control/HTTP.php

```diff
@@ -341,10 +341,13 @@ public static function add_cache_headers($body = null) {
 
 		$config = Config::inst()->forClass(__CLASS__);
 
+		// Get current cache control state, and clone it for modification
+		$cacheControl = clone HTTPCacheControl::singleton();
```

Comment:
This isn't really how it's meant to be; the singleton is global for a reason so that it can be modified or (at least) it's state queried in another method. There's nothing stopping multiple applications of the same header on a single request.

If you want to "reset" the state of the HTTPCacheControl you use injector as the docs state...

Reply:
I suppose we could then. I'll need to revert and update the tests.

Reply:
Wouldn't you agree that with this style, the code is less prone to error? You don't need to remember to reset the state in this case. It's probably more robust.